pkg/addon/templateagent/decorator.go

```diff
@@ -508,6 +512,60 @@ func (d *caBundleDecorator) decorate(pod *corev1.PodTemplateSpec) error {
 	return nil
 }
 
+type supportResource string
+
+const (
+	supportResourceDeployment supportResource = "deployments"
+	supportResourceDaemonset  supportResource = "daemonsets"
+)
+
+type resourceRequirementsDecorator struct {
+	privateValues addonfactory.Values
+	resource      supportResource // only support daemonsets, deployments for now
+	logger        klog.Logger
+}
+
+func newResourceRequirementsDecorator(logger klog.Logger, resource supportResource,
+	privateValues addonfactory.Values) podTemplateSpecDecorator {
+	return &resourceRequirementsDecorator{
+		resource:      resource,
+		privateValues: privateValues,
+	}
+}
+
+func (d *resourceRequirementsDecorator) decorate(name string, pod *corev1.PodTemplateSpec) error {
+	requirements, ok := d.privateValues[ResourceRequirementsPrivateValueKey]
+	if !ok {
+		return nil
+	}
+
+	regexRequirements, ok := requirements.([]addonfactory.RegexResourceRequirements)
+	if !ok {
+		return fmt.Errorf("resource requirements value is invalid")
+	}
+
+	for i := range pod.Spec.Containers {
```

Comment:
Should we consider `pod.Spec.InitContainers` as well?

Reply:
I am not quite sure, all other decorators do not consider the init containers.